FoundationsMastery/CustomString.cs

```diff
@@ -0,0 +1,62 @@
+﻿using FoundationsMastery.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FoundationsMastery
+{
+    public class CustomString : Interfaces.ICustomString
+    {
+        
+
+        public CustomString(char[] mychars)
+        {
+            Contents = mychars;
+        }
+
+        public CustomString()
+        {}
+
+        public IEnumerable<char> Contents
+        {
+            get; set;
+        }
+
+        public int Length
+        {
+            get;
+        }
+
+        public void Clear()
+        {
+            Contents = null;
+        }
+
+        public string Concat(IEnumerable<char> rhs)
+        {
+            return Contents.Concat(rhs).ToString();
+        }
+
+        public string Interleave(IEnumerable<char> rhs)
+        {
+            {
+                string valuePassed = String.Concat(rhs);
+                string contents = String.Concat(Contents);
+                string newString = "";
+                for (int i = 0; i < Contents.Count(); i++)
```

Comment:
This is fine for small strings... but consider how many strings we are creating and destroying if each string is 10,000 characters long. A `StringBuilder` instance is usually a good idea for these situations